app/validation/validator.py

```diff
@@ -1361,6 +1363,93 @@ def _validate_placeholder_transforms(self, transforms, block_id):
+                answer_label = answer['label']
+
+                if self._search_dumb_quotes(answer_label):
+                    errors.append(self._error_message(
+                        'Found dumb quotes in answer label/value for answer id {}'.format(answer['id'])))
+
+        for option in answer_options:
+
+            if isinstance(option['label'], str):
+                label = option['label']
+            else:
+                label = option['label']['text']
+
+            if 'description' in option:
+                if self._search_dumb_quotes(option['description']):
+                    errors.append(self._error_message(
+                        'Found dumb quotes in answer description for answer id {}'.format(answer['id'])))
+
+            if self._search_dumb_quotes(label) or \
+                    self._search_dumb_quotes(str(option['value'])) is not None:
+                errors.append(self._error_message(
+                    'Found dumb quotes in answer label/value for answer id {}'.format(answer['id'])))
+
+        self._validate_dumb_quotes_in_guidance(answer, errors, answer['id'])
+
+        return errors
+
+    def _validate_dumb_quotes_in_guidance(self, sub_block, errors, sub_block_id):
+        sub_block_guidance = sub_block.get('guidance', {})
+
+        if sub_block_guidance != {}:
+            if ('hide_guidance' in sub_block_guidance) and \
+                    (self._search_dumb_quotes(sub_block_guidance['hide_guidance'])):
+                errors.append(self._error_message(
+                    'Found dumb quotes in question/answer guidance for id {}'.format(sub_block_id)))
+            if ('show_guidance' in sub_block_guidance) and \
+                    (self._search_dumb_quotes(sub_block_guidance['show_guidance'])):
+                errors.append(self._error_message(
+                    'Found dumb quotes in question/answer guidance for id {}'.format(sub_block_id)))
+
+            sub_block_contents = sub_block_guidance.get('contents', {})
+            if ('title' in sub_block_contents[0]) and \
+                    (self._search_dumb_quotes(sub_block_contents[0]['title'])):
+                errors.append(self._error_message(
+                    'Found dumb quotes in question/answer guidance for id {}'.format(sub_block_id)))
+            if ('description' in sub_block_contents[0]) and \
+                    (self._search_dumb_quotes(sub_block_contents[0]['description'])):
+                errors.append(self._error_message(
+                    'Found dumb quotes in question/answer guidance for id {}'.format(sub_block_id)))
+            if 'list' in sub_block_contents[0]:
+                for item in sub_block_contents[0]['list']:
+                    if self._search_dumb_quotes(str(item)):
+                        errors.append(self._error_message(
+                            'Found dumb quotes in question/answer guidance for id {}'.format(sub_block_id)))
```

Comment:
In `eq-translations` we search for all text elements in a schema (have a look at the `no_context_pointers` and `context_pointers` methods). We should look to extract that into something common that we can use here.

Reply:
Refactored with code from eq-translations